### PR TITLE
Feature/add business logic of rule settings

### PR DESCRIPTION
**Pull request by shaun-ak**
PR for #3772

### CODE REVIEW

File: apps/client/src/app/components/rule/rule.component.ts [hunk 1]
```diff
@@ -58,10 +58,12 @@ export class RuleComponent implements OnInit {
       .subscribe(
         ({ settings }: { settings: PortfolioReportRule['settings'] }) => {
```
Comment (dtslvr) [inline apps/client/src/app/components/rule/rule.component.ts:59]: My bad, I get a deconstruction error here. Can you please change to

```
(settings: PortfolioReportRule['settings']) => {
```

File: apps/client/src/app/components/rule/rule-settings-dialog/rule-settings-dialog.html
Comment (dtslvr) [inline apps/client/src/app/components/rule/rule-settings-dialog/rule-settings-dialog.html:1]: Can you change the line

```
(click)="dialogRef.close({ settings })"
```

to 
```
(click)="dialogRef.close(settings)"
```

File: apps/api/src/app/user/update-user-setting.dto.ts [hunk 2]
```diff
@@ -88,6 +89,9 @@ export class UpdateUserSettingDto {
   @IsOptional()
   locale?: string;
 
+  @IsOptional()
+  portfolioSettings?: PortfolioReportRule['settings'];
```
Comment (dtslvr) [inline apps/api/src/app/user/update-user-setting.dto.ts:93]: Attention: data needs to be stored on `xRayRules` to make it work

You could adapt `RuleSettings` in `XRayRulesSettings` to something like this:

```
interface RuleSettings extends Pick<PortfolioReportRule, 'settings'> {
  isActive: boolean;
}
```

File: apps/client/src/app/components/rule/rule.component.ts [hunk 3]
```diff
@@ -58,10 +58,12 @@ export class RuleComponent implements OnInit {
...
           if (settings) {
-            console.log(settings);
-
-            // TODO
-            // this.ruleUpdated.emit(settings);
+            const updatedSettings: UpdateUserSettingDto = {
+              portfolioSettings: {
```
Comment (dtslvr) [inline apps/client/src/app/components/rule/rule.component.ts:62]: Change to something like this

```
xRayRules: {
  [rule.key]: settings
}
```

File: apps/client/src/app/components/rule/rule.component.ts [hunk 4]
```diff
@@ -58,10 +58,12 @@ export class RuleComponent implements OnInit {
...
+                ...settings
+              }
+            };
+            this.ruleUpdated.emit(updatedSettings);
```
Comment (dtslvr) [inline apps/client/src/app/components/rule/rule.component.ts:66]: Add line break

**Review by dtslvr — CHANGES_REQUESTED** [review]
Thank you for your contribution, @shaun-ak!

I have added a few comments. To test, you could customize _Threshold Max_ in _Fee Ratio_ to `0.05` and it should update the text to _The fees do not exceed 5% of your initial investment (0.00%)_.

**Comment by shaun-ak** [comment]
Hey @dtslvr , I tried to test it but the text does not change, however the correct object is emitted, I guess we need to update something on the api end?

**Review by dtslvr — APPROVED** [review]
> Hey @dtslvr , I tried to test it but the text does not change, however the correct object is emitted, I guess we need to update something on the api end?

Thank you for the contribution, @shaun-ak! 😎

I was able to finalize it, the only missing piece was providing the persisted setting (see [here](https://github.com/ghostfolio/ghostfolio/pull/3826/commits/b884ae3599ca174b22c9d2385cd827a700184eb3#diff-3c888bd2feb1b3eb21c716ab7d3ee77c3679a32922d79882b533dbfc211a6c31R83)).

Ready to merge now.

**Pull request merged**